client-java/instrumentation/src/main/java/org/evomaster/client/java/instrumentation/ClassAnalyzer.java

```diff
@@ -11,22 +11,84 @@
+    private static final String JAKARTA_FUTURE_OR_PRESENT_ANNOTATION_NAME = "jakarta.validation.constraints.FutureOrPresent";
+    private static final String JAVAX_NULL_ANNOTATION_NAME = "javax.validation.constraints.Null";
+    private static final String JAKARTA_NULL_ANNOTATION_NAME = "jakarta.validation.constraints.Null";
+    private static final String JAVAX_SIZE_ANNOTATION_NAME = "javax.validation.constraints.Size";
+    private static final String JAKARTA_SIZE_ANNOTATION_NAME = "jakarta.validation.constraints.Size";
+    private static final String JAVAX_DIGITS_ANNOTATION_NAME = "javax.validation.constraints.Digits";
+    private static final String JAKARTA_DIGITS_ANNOTATION_NAME = "jakarta.validation.constraints.Digits";
+    private static final String JAVAX_ENTITY_ANNOTATION_NAME = "javax.persistence.Entity";
+    private static final String JAKARTA_ENTITY_ANNOTATION_NAME = "jakarta.persistence.Entity";
+    private static final String JAVAX_TABLE_ANNOTATION_NAME = "javax.persistence.Table";
+    private static final String JAKARTA_TABLE_ANNOTATION_NAME = "jakarta.persistence.Table";
+
+    private static final String JAVAX_NOT_NULL_ANNOTATION_NAME = "javax.validation.constraints.NotNull";
+    private static final String JAKARTA_NOT_NULL_ANNOTATION_NAME = "jakarta.validation.constraints.NotNull";
+    public static final String JAVAX_COLUMN_ANNOTATION_NAME = "javax.persistence.Column";
+    public static final String JAKARTA_COLUMN_ANNOTATION_NAME = "jakarta.persistence.Column";
+    public static final String JAVAX_TRANSIENT_ANNOTATION_NAME = "javax.persistence.Transient";
+    public static final String JAKARTA_TRANSIENT_ANNOTATION_NAME = "jakarta.persistence.Transient";
+    public static final String JAVAX_ENUMERATED_ANNOTATION_NAME = "javax.persistence.Enumerated";
+    public static final String JAKARTA_ENUMERATED_ANNOTATION_NAME = "jakarta.persistence.Enumerated";
+
     /**
      * Try to load the given classes, and do different kind of analysis via reflection.
      * This can be helpful when doing such analysis at bytecode level (eg when intercepting class loader
      * with instrumentator) would be bit complex
      * <p>
      * Note: this will have side-effects on static data-structures
      *
-     * @param classNames
+     * @param classNames the list of names of classes to be analyzed
      */
     public static void doAnalyze(Collection<String> classNames) {
 
-        boolean jpa = canUseJavaxJPA();
+        // Java Persistence API
+        boolean canUseJavaPersistenceApi = canUseJavaxJPA();
+
+        // Jakarta Persistence
+        boolean useJakartaPersistenceLayer;
+        if (canUseJavaPersistenceApi) {
+            // If JPA can be used, then JPL cannot be used
+            useJakartaPersistenceLayer = false;
+        } else {
+            // Check if Jakarta Persistence Layer can be used
+            useJakartaPersistenceLayer = canUseJakartaPersistenceLayer();
```

Comment:
naming should be consistent. if we have `canUseJavaPersistenceApi`, then I guess should be `canUseJakartaPersistenceLayer`?

Reply:
my bad.